connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java

```diff
@@ -222,6 +223,20 @@ public void stop() {
         connectorStatusMetricsGroup.close();
 
         workerConfigTransformer.close();
+        executor.shutdown();
+        try {
+            // Wait a while for existing tasks to terminate
+            if (!executor.awaitTermination(EXECUTOR_SHUTDOWN_TERMINATION_TIMEOUT_MS, TimeUnit.MILLISECONDS)) {
+                executor.shutdownNow(); //cancel current executing threads
+                // Wait a while for tasks to respond to being cancelled
+                if (!executor.awaitTermination(EXECUTOR_SHUTDOWN_TERMINATION_TIMEOUT_MS, TimeUnit.MILLISECONDS))
+                    log.error("Pool did not terminate");
```

Comment:
nit: I think we should improve this message a little bit. How about:
`Executor did not terminate in time.`

Reply:
corrected the message